includes/subscriptions/class-wc-payments-product-service.php

```diff
@@ -128,11 +128,12 @@ public function get_wcpay_product_id( WC_Product $product, $test_mode = null ) :
 	 * return string       The item's WCPay product id.
 	 */
 	public function get_wcpay_product_id_for_item( string $type ) : string {
-		if ( ! get_option( self::get_wcpay_product_id_option() . '_' . $type ) ) {
-			$this->create_product_for_item_type( $type );
+		$sanitized_type = sanitize_key( str_replace( ' ', '_', trim( strtolower( $type ) ) ) );
```

Comment:
A micro-optimisation suggession: `strtolower( trim( $type ) )`

Reply:
Thanks for reviewing, and those very helpful early learnings for me. I have made the changes in the update 
https://github.com/Automattic/woocommerce-payments/pull/4496/commits/a891e1437fe8adf3adeb5ee77684d58acf24c453#

I have unit tests planned for this. Being the first time, I am taking a bit longer to bring it in :D